pynicotine/plugins/plugin_debugger/__init__.py

```diff
@@ -26,114 +28,131 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
 
+        self.settings = {}
+        self.metasettings = {}
+
+        for method_name, method in getmembers(self, ismethod):
+            if method.__module__ != __name__ or method_name.startswith("_"):
+                # Convenience function from BasePlugin, don't include these
+                continue
+
+            self.settings[method_name] = ("Verbose" not in getsource(method))
+            self.metasettings[method_name] = {
+                "description": f"{method_name}()",
+                "type": "bool"
+            }
+
         self.log("__init__()")
 
+    def _trigger_log(self, parameter_arguments=""):
```

Comment:
It should be possible to turn this into a decorator that extracts the function names and arguments automatically.

Reply:
Yes but the more tricky part is binding the live values of each argument. To do that is going to result in quite a lot of code and is beyond my current level of understanding.